### DISCUSSION TITLE
v1.1-BETA-20220330

### DISCUSSION THREAD

**Discussion by erichelgeson** [erichelgeson/BlueSCSI-v1#79] [General]
# This is a pre-release!

1. Please backup your images before testing, just in case. 
1. Please report issues, compatibility improvements or changes on the Github discussion
   Before reporting an issue please include the BlueSCSI hardware rev, computer tested on, image used, and re-test after doing a full overwrite format with "SD Card Formatter"
1. ETA on stable release is TBD.

# Changes

* Apple HD SC Setup support #71
  * You can now format in most Mac utilities without error.
* Image set selection #73 
  * Add a physical switch/jumpers externally to your BlueSCSI so you can switch computers without moving files on the SD card. Select between 4 image sets (folders of images) on the SD card. There is also an optional shared image set.
  * See Github issue for usage instructions. Will be on wiki once released.
* CLZ optimization #63, Optimize handling of reset interrupt. #64, Code reordering for bus settle delays #78, Upgrade to latest toolchain #67, #70
  * These changes give a performance improvement of 10%+ and better stability. My 6100 does 1.2mb/sec read, 1.2mb/sec write.
* Be more lenient on SD cards that are not able to communicate at 50mhz SPI #69
  * Some SD Cards would fail to communicate with the BlueSCSI at 50mhz but work fine at 49mhz. No real change in performance till you reach 40mhz, which a warning will be logged.
* Fixes to get VAXstation 3100 working #59

Compatible with Hardware versions 1.0-a, b, c and 1.1-a.

# Contributors

@mactcp, @pacjunk, @erichelgeson,  @marcelv-3 

# Should I upgrade?

If you want to be part of testing features before it's released and are willing to report your findings, go for it!

# Flashing

To update you will need a STLink V2 clone programmer and follow the flashing instructions here https://github.com/erichelgeson/BlueSCSI/wiki/Flashing#using-stlink-v2-preferred

<hr /><em>This discussion was created from the release <a href='https://github.com/erichelgeson/BlueSCSI/releases/tag/v1.1-BETA-20220330'>v1.1-BETA-20220330</a>.</em>

**Comment by marcelv-3** [comment]
tested this version in my se/30 system 7.0.1 bluescsi 1.1
works great and the read and write speed have increased by 10 %
will do some more testing on other models and configs

**Comment by erichelgeson** [comment]
I've gone through this procedure with this release https://github.com/erichelgeson/BlueSCSI/wiki/Manual-Testing---Dev and seems good. We'll cut a release tomorrow if no issues reported.

**Reply by marcelv-3** [reply]
testen also with a LC III with internal blue scsi with system 7.5.5 image and at the same time a DB25 version connected to the external port with system 7.0.1 image
boot from internal, and use silver lining 5.31 Time Drive from the external drive

timings:
external v1.1-20211215
- Write 641 KBytes
- Read 1048 KBytes

internal v1.1-20211215
- Write 714 KBytes
- Read 1048 KBytes


external v1.1-BETA-20220330
- Write 748 KBytes
- Read 1258 KBytes

internal v1.1-BETA-20220330
- Write 850 KBytes
- Read 1258 KBytes


also tested with a large rascsi image, speeds are similar, only the latency becomes very high

no issues in my tests
====================================
tested as external on Color Classic System 7.1
- Write 953 KBytes
- Read 1084 KBytes

**Comment by baldengineer** [comment]
40 and 50 millihertz? How long do transfers take at such a slow speed?!

**Reply by erichelgeson** [reply]
Did you experience slow speeds at the slower SPI? What SD Card did you use and can you share the log? In my testing there was no change. If you have a suggestion on how to configure the SPI and be lenient on different cards please let us know. The current default(max) is 50 megahertz.

**Reply by Tyrantulas** [reply]
I thought the thing about pedantry was that you had to actually be right about the thing you're trying to correct the other person on?

"mHz" is millihertz.  "mhz" is "I'm busy living my life, AND trying by best to run a community project that is appreciably helping retrocomputing fans as my time and abilities allow.  I'm not perfect, and don't ask anyone else to be either."

Something like that?

https://en.wikipedia.org/wiki/Hertz#SI_multiples

**Comment by MrGasS** [comment]
I'm testing this firmware on my homemade BlueSCSI board, based on a legit BluePill.
My first gen LC is running fine with it and it's fast as hell, I installed the 7.1.2 version of MacOS, played few games and opened few Word documents, no problems at all.
Apple SD HC Setup formatted my 2 GB hardfile successfully.
I don't want to take any speed test because I'm using an old Class 4 microSD card from 10 years ago formatted in exFAT, but everything works fine.

**Reply by erichelgeson** [reply]
Thanks for testing @MrGasS !

**Reply by MrGasS** [reply]
Thank you for this wonderful job, BlueSCSI is amazing.
